common/check_internal.h

```diff
@@ -14,6 +14,11 @@ namespace Carbon::Internal {
 //
 // This is out-of-line and will arrange to stop the program, print any debugging
 // information and this string.
+//
+// This API uses `const char*` C string arguments rather than `llvm::StringRef`
+// because that lets the code size of calling it be smaller as it only needs to
```

Comment:
```suggestion
// because that lets the code size of calling it be smaller: it only needs to
```

Huh, so even though there's 4 pointers, it materializes all of that as a single pointer?

Reply:
No, I mean that if we make these `llvm::StringRef`s, then it has to materialize both a pointer and a size far that argument intsead of just a pointer.

Reply:
can add "for each" to the sentence?

Reply:
Tried a better comment to really spell out what is different about the first arguments from the last. Sorry if that was the source of confusion and I didn't address it at first.

Reply:
Ah, so this is just that sizeof(const char*) < sizeof(StringRef), okay.

Reply:
Am I awful to ask if `int line` should be `int32_t line`?

Reply:
The size of int won't matter here, which is why I just used the general `int`.

Could change extra message but it would actually make it a tiny bit more constraining to call. Seemed better to use the general type there as only one.

Reply:
Going to merge while this is clean, but happy to follow up on this if useful.